test/unified_format.py

```diff
@@ -814,84 +814,79 @@ def match_result(self, expectation, actual, in_recursive_call=False):
+            if field == "type":
+                field = "server_type_name"
+            self.test.assertEqual(getattr(actual, field), expected)
+
+    def match_topology_description(self, actual: TopologyDescription, spec: dict) -> None:
+        for field, expected in spec.items():
+            field = camel_to_snake(field)
+            if field == "type":
+                field = "topology_type_name"
+            self.test.assertEqual(getattr(actual, field), expected)
+
+    def match_event_fields(self, actual: Any, spec: dict) -> None:
+        for field, expected in spec.items():
+            if field == "command" and isinstance(actual, CommandStartedEvent):
+                command = spec["command"]
+                if command:
+                    self.match_result(command, actual.command)
+                continue
+            if field == "reply" and isinstance(actual, CommandSucceededEvent):
+                reply = spec["reply"]
+                if reply:
+                    self.match_result(reply, actual.reply)
+                continue
+            if field == "hasServiceId":
+                if spec["hasServiceId"]:
+                    self.test.assertIsNotNone(actual.service_id)
+                    self.test.assertIsInstance(actual.service_id, ObjectId)
+                else:
+                    self.test.assertIsNone(actual.service_id)
+                continue
+            if field == "hasServerConnectionId":
+                if spec["hasServerConnectionId"]:
+                    self.test.assertIsNotNone(actual.server_connection_id)
+                    self.test.assertIsInstance(actual.server_connection_id, int)
+                else:
+                    self.test.assertIsNone(actual.server_connection_id)
+                continue
+            if field in ("previousDescription", "newDescription"):
+                if isinstance(actual, ServerDescriptionChangedEvent):
+                    self.match_server_description(
+                        getattr(actual, camel_to_snake(field)), spec[field]
+                    )
+                    continue
+                if isinstance(actual, TopologyDescriptionChangedEvent):
+                    self.match_topology_description(
+                        getattr(actual, camel_to_snake(field)), spec[field]
+                    )
+                    continue
 
-    def assertHasServerConnectionId(self, spec, actual):
-        if "hasServerConnectionId" in spec:
-            if spec.get("hasServerConnectionId"):
-                self.test.assertIsNotNone(actual.server_connection_id)
-                self.test.assertIsInstance(actual.server_connection_id, int)
+            if field == "interruptInUseConnections":
+                field = "interrupt_connections"
             else:
-                self.test.assertIsNone(actual.server_connection_id)
+                field = camel_to_snake(field)
+            self.test.assertEqual(getattr(actual, field), expected)
```

Comment:
I refactored this code because the old version ignored any unknown fields. The new version fails on unknown fields. It also should be easier to maintain since fields that only need to be renamed with `camel_to_snake` are supported automatically like "minWireVersion", "topologyVersion", "commandName", etc....